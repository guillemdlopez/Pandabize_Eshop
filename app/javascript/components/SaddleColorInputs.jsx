import React from "react";
import { borderColor } from "../fixtures/functions";
import PropTypes from "prop-types";

const SaddleColorInputs = ({
  saddleColors,
  handleInputChange,
  setPrice,
  setPriceSaddle,
  price,
  setErrorMsg,
  displayError,
  errorMsgContent,
}) => {
  return (
    <>
      <p className="text-muted text-center mb-0">Saddle color</p>
      <ul
        className="block-list-center customizations-list text-center animate__animated animate__fadeIn"
        id="custom-associations"
      >
        {saddleColors.map((color) => {
          return (
            <li
              key={color.id}
              id="saddle-colors"
              style={borderColor(color.value)}
            >
              <label htmlFor={color.id} className="mb-0 mr-2">
                {color.value}
              </label>
              <input
                type="radio"
                name="saddleColor"
                key={color.id}
                value={color.id}
                onClick={(e) => {
                  handleInputChange(e);
                  setPriceSaddle(color.price);
                  setPrice(price + color.price);
                  setErrorMsg({ displayError: false, errorMsgContent: "" });
                }}
                data-id={color.id}
              />
            </li>
          );
        })}
      </ul>
    </>
  );
};

SaddleColorInputs.propTypes = {
  saddleColors: PropTypes.array.isRequired,
  handleInputChange: PropTypes.func.isRequired,
  setPrice: PropTypes.func.isRequired,
  setPriceSaddle: PropTypes.func.isRequired,
  price: PropTypes.number.isRequired,
  setErrorMsg: PropTypes.func.isRequired,
  displayError: PropTypes.bool.isRequired,
  errorMsgContent: PropTypes.string.isRequired,
};

export default SaddleColorInputs;
