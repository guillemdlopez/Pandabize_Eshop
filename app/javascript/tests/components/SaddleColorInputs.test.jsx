import { shallow } from "enzyme";
import React from "react";
import SaddleColorInputs from "../../components/SaddleColorInputs";

describe("Testing the <SaddleColorInputs/> component", () => {
  const saddleColorInputsProps = {
    saddleColors: [
      {
        id: 191,
        name: "Saddle color",
        value: "Yellow",
        price: 7.11,
        bicycle_id: 26,
      },
      {
        id: 193,
        name: "Saddle color",
        value: "Black",
        price: 8.36,
        bicycle_id: 26,
      },
    ],
    handleInputChange: jest.fn(),
    setPrice: jest.fn(),
    setPriceSaddle: jest.fn(),
    price: 0,
    setErrorMsg: jest.fn(),
    displayError: false,
    errorMsgContent: "",
  };
  const wrapper = shallow(<SaddleColorInputs {...saddleColorInputsProps} />);
  test("should render the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should display a label and an input for each saddle color", () => {
    const { saddleColors } = saddleColorInputsProps;
    const values = saddleColors.map((color) => color.value);

    const [yellow, black] = values;
    expect(wrapper.find("label").at(0).text()).toBe(yellow);
    expect(wrapper.find("label").at(1).text()).toBe(black);
    expect(wrapper.find("input").length).toBe(values.length);
  });
});
