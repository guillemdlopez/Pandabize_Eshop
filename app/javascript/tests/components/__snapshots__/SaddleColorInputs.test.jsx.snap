// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <SaddleColorInputs/> component should render the component correctly 1`] = `
<Fragment>
  <p
    className="text-muted text-center mb-0"
  >
    Saddle color
  </p>
  <ul
    className="block-list-center customizations-list text-center animate__animated animate__fadeIn"
    id="custom-associations"
  >
    <li
      id="saddle-colors"
      key="191"
      style={
        Object {
          "border": "2px solid rgb(255, 255, 0)",
          "display": "inline-block",
          "marginRight": "10px",
        }
      }
    >
      <label
        className="mb-0 mr-2"
        htmlFor={191}
      >
        Yellow
      </label>
      <input
        data-id={191}
        key="191"
        name="saddleColor"
        onClick={[Function]}
        type="radio"
        value={191}
      />
    </li>
    <li
      id="saddle-colors"
      key="193"
      style={
        Object {
          "border": "2px solid rgb(0, 0, 0)",
          "display": "inline-block",
          "marginRight": "10px",
        }
      }
    >
      <label
        className="mb-0 mr-2"
        htmlFor={193}
      >
        Black
      </label>
      <input
        data-id={193}
        key="193"
        name="saddleColor"
        onClick={[Function]}
        type="radio"
        value={193}
      />
    </li>
  </ul>
</Fragment>
`;
